### PR TITLE
Repo signature rework

### PR DESCRIPTION
**Pull request by ydirson**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -60,14 +127,31 @@ This scripts excludes from the mirror:
...
+> this includes much more packages than needed for producing
+> the install ISO, notably build-dependencies that are not needed by
+> the installer, and not get installed themselves on the XCP-ng host
+> either.  But a local mirror which you control will provide image
+> reproducibility, and ability to work offline.
+
+> [!NOTE]
+>
+> this is not a general-purpose mirroring tool.  It will
+> notably not mirror a number of deveopment packages, which you owuld
```
Comment (stormi) [inline README.md]:
```suggestion
> notably not mirror a number of development packages, which you would
```

**Review by stormi — COMMENTED** [review]
LGTM at this stage (I only checked the readme)

File: README.md (hunk 1 above)
Comment (ydirson) [reply README.md]: done

**Comment by ydirson** [comment]
Remaining issue is that the yum repo is set up in such a way that a single GPG key has to be used for both `repo-gpg-check` (checking the sig on repo metadata, which is the sig on which this PR focuses) and for `gpg-check` (checking the sig on individual RPMs).
My position is that the `gpg-check=1` setting is superfluous anyway: the sig on the repo metadata encompasses everything in the repo, down to the RPMs themselves. We could simply change it `gpg-check=0` without sacrificing anything security-wise with our install media (provided they checked the signature of the image itself before booting it, as it is the source of the public key used to verify the repo when doing netinstall).  Someone building a custom media ought to sign their repo metadata and gain the same level of integrity. If they don't (but, again, they really should), they can only get some level of integrity for netinstall through `gpg-check=1`, and this only if they add RPMs that are already signed by the key provided on the ISO -- so really this is a tiny niche, and they should sign their metadata instead.

File: README.md [hunk 2]
```diff
@@ -59,15 +95,55 @@ This scripts excludes from the mirror:
 - source RPMs
 - development RPMs
 - debugging-symbols RPMs
+- a few large RPMs only useful as build dependencies: ocaml, golang,
+  java
```
Comment (stormi) [inline README.md:99]:
```suggestion
  java, xs-opam-repo, etc.
```

Comment (ydirson) [reply README.md:99]: I considered `xs-opam-repo` to be part of `ocaml`

Comment (stormi) [reply README.md:99]: ok

**Review by stormi — COMMENTED** [review]
It looks good to me overall.

Minor regression: `create-iso.sh` with `--netinstall` adds a `.treeinfo` file (whose placeholders were not replaced)

**Comment by ydirson** [comment]
> Minor regression: `create-iso.sh` with `--netinstall` adds a `.treeinfo` file (whose placeholders were not replaced)

Good catch, fixing that one

**Comment by ydirson** [comment]
Only one issue remain (for signing with any other key than the XCP-ng one):
- boot with `no-gpgcheck` by default?
- keep it and tell the users to do that themselves (scripted in sample sign script)?
- dig further do find how to separate the repo and rpm sig checking, which may imply changes to host-installer?
- or?

**Comment by stormi** [comment]
> Only one issue remain (for signing with any other key than the XCP-ng one):
> 
>     * boot with `no-gpgcheck` by default?
>     * keep it and tell the users to do that themselves (scripted in sample sign script)?
>     * dig further do find how to separate the repo and rpm sig checking, which may imply changes to host-installer?
>     * or?

I like the idea to let the sample script disable gpgcheck. But then, will signing the repodata serve any purpose? 

Best option in the long term is probably option 3, but we can let it wait for when either users or ourselves express a need for it.

**Comment by ydirson** [comment]
> I like the idea to let the sample script disable gpgcheck. But then, will signing the repodata serve any purpose?

You indeed close the door on the idea here :smile: 
I was thinking 8.3 where we have separate controls for `gpgcheck` and `repo_gpgcheck`, but the original xcpng patches in 8.2 have a single control for both, so that cannot work on our LTS version.

That would require the sample script to behave differently depending on the version, that's IMHO not the place for it.

> Best option in the long term is probably option 3, but we can let it wait for when either users or ourselves express a need for it.

Until we can dedicate some time for 3., that seems to leave only 1., with a new config file in `create-iso` to let 8.3 pass `no-gpgcheck` and not modify the 8.2 behavior (after all, 8.2 never supported re-signing the repo without resigning the RPMs, it seems).

**Comment by ydirson** [comment]
Added the mechanism for 8.3 to use `no-gpgcheck`

**Comment by stormi** [comment]
LGTM. I'll generate a signed ISO one last time before I approve the PR.

@gduperrey as planned by Yann and myself, this pull request changes the way we call the generation scripts somehow. Not as much as we initially planned, but maybe enough to require adapting the CI jobs that generate daily (unsigned) ISOs. Please test them and adapt them if needed, with Yann's help.

Once you have Before/After ISOs for both 8.2 and 8.3, I'll compare them and look for unexpected changes.

Then we'll merge this PR.

**Comment by gduperrey** [comment]
@stormi I have a PR pending.
As Jenkins is using a local copy, I will copy/manage it when the PRs will be merged.

**Comment by stormi** [comment]
Unless I've overlooked it, I think the README is lacking an actual example of invocation of `create-iso.sh` using a signing script.

**Review by stormi — CHANGES_REQUESTED** [review]
I generated a 8.3:updates ISO, signed, but the bootloader entries all contain `no-gpgcheck`.

**Comment by ydirson** [comment]
Changed default behavior not to add `no-gpgcheck` and let users add it in their own layer when needed.

**Review by stormi — APPROVED** [review]

**Pull request merged**